sdk/cosmos/azure-cosmos-tests/src/test/java/com/azure/cosmos/ProactiveConnectionManagementTest.java

```diff
@@ -636,14 +742,50 @@ private Object[] proactiveContainerInitConfigs() {
                 .withContainerCount(3)
                 .withMinConnectionPoolSizePerEndpoint(4)
                 .withAggressiveWarmupDuration(Duration.ofMillis(250))
-                .withIsSystemPropertySetBeforeDirectConnectionConfig(true),
+                .withIsSystemPropertySetBeforeDirectConnectionConfig(true)
+                .withIsSyncClient(false),
+            new ProactiveConnectionManagementTestConfig()
+                .withPreferredRegions(preferredRegions)
+                .withProactiveConnectionRegionsCount(2)
+                .withContainerCount(3)
+                .withMinConnectionPoolSizePerEndpoint(5)
+                .withAggressiveWarmupDuration(Duration.ofMillis(1000))
+                .withIsSystemPropertySetBeforeDirectConnectionConfig(false)
+                .withIsSyncClient(false),
+            // in this test config the aggressive warmup duration is set to an unreasonably high value
+            // the client build should not block until this aggressive warm up duration time
+            new ProactiveConnectionManagementTestConfig()
+                .withPreferredRegions(preferredRegions)
+                .withProactiveConnectionRegionsCount(2)
+                .withContainerCount(3)
+                .withMinConnectionPoolSizePerEndpoint(5)
+                .withAggressiveWarmupDuration(Duration.ofMinutes(1000))
+                .withIsSystemPropertySetBeforeDirectConnectionConfig(false)
+                .withIsSyncClient(false),
+            new ProactiveConnectionManagementTestConfig()
```

Comment:
since we introduced faultInjection on gateway, can we also add test cases where some errors during the warm up stage - for example collection read timed out etc. - can be in a different PR

Reply:
Tracking the work item here - https://github.com/Azure/azure-sdk-for-java/issues/33088